src/main/java/emu/grasscutter/game/quest/GameMainQuest.java

```diff
@@ -120,9 +124,233 @@ public void finish() {
         // handoff main quest
         if(mainQuestData.getSuggestTrackMainQuestList() != null){
             Arrays.stream(mainQuestData.getSuggestTrackMainQuestList())
-                .forEach(getOwner().getQuestManager()::startMainQuest);
+                .forEach(getQuestManager()::startMainQuest);
         }
 	}
+    //TODO
+    public void fail() {}
+    public void cancel() {}
+
+    public void tryAcceptSubQuests(QuestTrigger condType, String paramStr, int... params) {
+        try {
+            SubQuestData[] subQuestsData = GameData.getMainQuestDataMap().get(this.parentQuestId).getSubQuests();
+            ArrayList<QuestData> subQuests = new ArrayList<>();
+            for(SubQuestData subQuestData: subQuestsData) {
+                subQuests.add(GameData.getQuestDataMap().get(subQuestData.getSubId()));
+            }
+            /*
+                Make sure we check the subQuests with the exact condition AND parameters
+                Note: because we currently don't pass all parameters (example: QUEST_CONTENT_ENTER_DUNGEON normally has 2 parameters),
+                I'll only check for params[0], which is a worse filter
+            */
+            List<QuestData> subQuestsWithCond = subQuests.stream()
```

Comment:
also, i think it is unnecessary. when it comes to a condition which is not meet or not impl, the trigger will just return `false` so that the quest will not be accepted.

Reply:
I agree. but some QuestTriggers (example: QUEST_CONTENT_GAME_TIME_TICK) have same params ,but different paramStr.
I'll make it `q.getParam()[0] == params[0] && q.getParamStr == paramStr`

Reply:
i think it is hardcode. if they have the same params, they should all get triggered. It wont cause any exceptions when more than one quest progress change at the same time

Reply:
Looking at QuestExcelConfigData again, there area lot of different conditions with `params` = [0,0]. I think triggering many handlers that we know are going to return false is going to scale very poorly with a larger playerbase, who each  might have several quests active. Isn't filtering beforehand more computationally efficient? maybe we should check `condition` and `params` and `paramStr`

Reply:
but the problem is the conditions are not always `==`, like `QUEST_COND_PLAYER_LEVEL_EQUAL_GREATER` it is `>=`

Reply:
indeed... also in `QUEST_CONTENT_GAME_TIME_TICK` we are given a range. It seems without adding special cases, the simplest filter is ` q.getType() == condType`